### PR TITLE
Update the CodeQL configuration

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request pulls in changes to the CodeQL workflow based on https://github.com/cisagov/skeleton-python-library/pull/98.

## 💭 Motivation and context ##

This gets the CodeQL configuration in this project (and its descendants) up-to-date.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -11,59 +10,58 @@ on:
...
+      # required for all workflows
+      security-events: write
     strategy:
       fail-fast: false
       matrix:
         # Override automatic language detection by changing the below list
-        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript',
-        # 'python']
-        language: ['python']
+        # Supported options are go, javascript, csharp, python, cpp, and java
```
Comment (dav3r) [inline .github/workflows/codeql-analysis.yml:33]: Looks like we missed alphabetizing this in [skeleton-python-library](https://github.com/cisagov/skeleton-python-library/blob/develop/.github/workflows/codeql-analysis.yml#L33).  Or is there a reason for this particular ordering of options?
```suggestion
        # Supported options are cpp, csharp, go, java, javascript, and python
```

Comment (mcdonnnj) [reply .github/workflows/codeql-analysis.yml:33]: The workflow is in large part cribbed from the [example workflow](https://github.com/github/codeql-action#usage) for the Action. I mostly just made some minor changes to the YAML in line with our usual preferences.

Comment (dav3r) [reply .github/workflows/codeql-analysis.yml:33]: So does that mean that you are against alphabetizing this list here and in [skeleton-python-library](https://github.com/cisagov/skeleton-python-library/blob/develop/.github/workflows/codeql-analysis.yml#L33)?

Comment (mcdonnnj) [reply .github/workflows/codeql-analysis.yml:33]: I think unless we did heavier editing I'd prefer to leave it as close as possible to the example for now.

Comment (dav3r) [reply .github/workflows/codeql-analysis.yml:33]: OK, no problem.

**Review by dav3r — APPROVED** [review]

**Pull request merged**